hearthstone/simulator/core/hero_pool.py

```diff
@@ -1021,5 +1037,37 @@ def hero_power_impl(self, context: 'BuyPhaseContext', board_index: Optional['Boa
         context.owner.coins += 1
 
 
+class Galakrond(Hero):
+    base_power_cost = 1
+    power_target_location = [CardLocation.STORE]
+
+    def hero_power_valid_impl(self, context: 'BuyPhaseContext', board_index: Optional['BoardIndex'] = None,
+                              store_index: Optional['StoreIndex'] = None):
+        return bool(context.owner.store)
+
+    def hero_power_impl(self, context: 'BuyPhaseContext', board_index: Optional['BoardIndex'] = None,
+                        store_index: Optional['StoreIndex'] = None):
+        store_minion = context.owner.pop_store_card(store_index)
+        context.owner.tavern.deck.return_cards(store_minion.dissolve())
+        higher_tier_minions = [card for card in context.owner.tavern.deck.unique_cards() if
+                               card.tier == min(store_minion.tier + 1, 6)]
+        discovered_minions = []
+        for _ in range(3):
+            if higher_tier_minions:
+                minion = context.randomizer.select_discover_card(higher_tier_minions)
+                higher_tier_minions.remove(minion)
+                discovered_minions.append(minion)
+        self.discover_queue.append(discovered_minions)
```

Comment:
Why do we have a separate discover queue attached to the hero, instead of the player, since it is just discovering minion cards?  Is it because the regular discover queue will return the unselected cards to the deck (and this hero specifically discovers cards that aren't in the deck?).  If that's the case, I think we should instead modify the way the discover queue on the player works, so that it also tracks what should be done with the undiscovered minions (return them to the deck, or destroy them).

Reply:
Unfortunately some heros can discover non-cards, such as players or spells. Additionally, this specific hero adds the discovered card to the store, which is not the default behavior of the player's discover action, which adds them to the player's hand. Some heros (but not this one) do discover cards not in the deck.

When dealing with discovered monster cards, we could definitely pass more arguments to deal with locations and whatnot, but I'm not sure I have time to figure this out at the moment. Also still not sure how we could deal with non-cards.

Reply:
It also gets complicated because a hero power may have a discover that operates differently from normal discovers, such as triple rewards, that can both occur in the same game.

Reply:
Thinking about it some more, one solution may be to pass an object into the discover queue that holds all this information (type of thing to discover, what to do with discovered thing, what to do with other options)